pkg/node/address.go

```diff
@@ -297,44 +288,41 @@ func SetIPv4(ip net.IP) {
 // - other IP address type.
 // It must be reachable on the network.
 func GetIPv4() net.IP {
-	addrsMu.RLock()
-	defer addrsMu.RUnlock()
-	return clone(addrs.ipv4Address)
+	n := localNode.Get()
+	return clone(n.GetNodeIP(false))
 }
 
 // SetInternalIPv4Router sets the cilium internal IPv4 node address, it is allocated from the node prefix.
 // This must not be conflated with k8s internal IP as this IP address is only relevant within the
 // Cilium-managed network (this means within the node for direct routing mode and on the overlay
 // for tunnel mode).
 func SetInternalIPv4Router(ip net.IP) {
-	addrsMu.Lock()
-	addrs.ipv4RouterAddress = clone(ip)
-	addrsMu.Unlock()
+	localNode.Update(func(n *types.Node) {
+		n.SetCiliumInternalIP(ip)
+	})
 }
 
 // GetInternalIPv4Router returns the cilium internal IPv4 node address. This must not be conflated with
 // k8s internal IP as this IP address is only relevant within the Cilium-managed network (this means
 // within the node for direct routing mode and on the overlay for tunnel mode).
 func GetInternalIPv4Router() net.IP {
-	addrsMu.RLock()
-	defer addrsMu.RUnlock()
-	return clone(addrs.ipv4RouterAddress)
+	n := localNode.Get()
+	return n.GetCiliumInternalIP(false)
 }
 
 // SetK8sExternalIPv4 sets the external IPv4 node address. It must be a public IP that is routable
 // on the network as well as the internet.
 func SetK8sExternalIPv4(ip net.IP) {
```

Comment:
nit: rename `SetK8sExternalIPv4()` to `SetNodeExternalIPv4()` (or viceversa) for consistency?

Reply:
I wasn't renaming these on purpose even though I was quite annoyed by the inconsistencies as a follow-up PR would remove these completely in favor of a LocalNodeInitializer.